### ISSUE TITLE
Setting a Display Text's Interpolation duration actually sets the line width

### ISSUE THREAD

**Issue by JakeGBLP** [PaperSpigot/Paper#8990]
### Expected behavior

When using Display#setInterpolationDuration() with a Display Text, the line width is set instead, works fine with other display entities.

### Observed/Actual behavior

Checked the NBT, interpolation duration is 0 by default, set it to 1 with the above mentioned method, line width changes to 1 and interpolation duration remains 0.

### Steps/models to reproduce

Spawn a Display Text entity, assert that its interpolation is 0, change it to another integer using the method, watch as the line width changes to the integer used in the method (assuming the entity has text and the integer < length of said text), check the interpolation again and notice how it hasn't changed.
Repeat those steps with other display entities and notice how the method's behavior is correct with those.

### Plugin and Datapack List

[22:25:53 INFO]: fServer Plugins (4):
[22:25:53 INFO]: #ed8106Bukkit Plugins:
[22:25:53 INFO]:  8- aLuskr, aSkBeer, aSkriptr, askript-reflect

-

[22:26:10 INFO]: There are 2 data packs enabled: a[7vanilla (built-in)a]7, a[7file/bukkit (world)a]
[22:26:10 INFO]: There are no more data packs available

### Paper version

[22:25:13 INFO]: foChecking version, please wait...
[22:25:13 INFO]: fThis server is running Paper version git-Paper-451 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 85a67da)r
aYou are running the latest versionr
7oPrevious version: git-Paper-"4f933b2" (MC: 1.19.4)

### Other

I have run this testing using Skript and Skript-Reflect but also through Lusk (self made Skript addon), result stays the same.

**Comment by lynxplay** [comment]
I am having trouble replicating this. Interpolation duration is set correctly for me.
Can you share specific examples using the API instead of a third party scripting engine ? I sadly am not too informed about skript to debug that myself.

**Comment by J-ak-e** [comment]
```import:
  org.bukkit.entity.Display

command /test:
  trigger:
    broadcast target.getInterpolationDuration()
    target.setInterpolationDuration(1)
    broadcast target.getInterpolationDuration()```
This snippet uses Skript and Skript-reflect, it takes the entity you're looking at and broadcasts the duration, then it sets it to 1 and then it broadcasts it again.

**Comment by J-ak-e** [comment]
I haven't made a standalone plugin in months, sorry. I just noticed this inconsistency and reported it. The way I'm using Skript to do it is also pretty reliable.

**Comment by lynxplay** [comment]
No worries :+1: 
```java
@EventHandler
public void on(final PlayerDropItemEvent event) {
    final Player player = event.getPlayer();

    player.getWorld().spawn(player.getEyeLocation(), TextDisplay.class, t -> {
        player.sendMessage(Component.text(t.getInterpolationDuration()));
        t.setInterpolationDuration(10);
        player.sendMessage(Component.text(t.getInterpolationDuration()));
    });
}
```

works fine for me here, the entity spawned also has the correct nbt data.
I presume something on skript is messing up ? Concerning I don't know that tho, I'll leave it open for someone with better  knowledge about skript.

**Comment by J-ak-e** [comment]
👍

**Comment by APickledWalrus** [comment]
Hello, I was working with Jake to determine the cause of the issue.

My initial test was conducted on a small test server with a few various plugins. I was able to reproduce the issue. Admittedly, not a very "clean" environment, so I performed some additional tests.

My next test was to remove all plugins and just use a simple testing plugin (paper-plugin specifically), similar to the provided above. Here is the code I used:
```java
@EventHandler
public void on(final PlayerDropItemEvent event) {
	final Player player = event.getPlayer();

	// before actual spawn
	player.getWorld().spawn(player.getEyeLocation(), TextDisplay.class, display -> {
		player.sendMessage(Component.text(display.getInterpolationDuration()));
		display.setInterpolationDuration(10);
		player.sendMessage(Component.text(display.getInterpolationDuration()));
	});

	// spawn but don't modify until after
	final TextDisplay display = player.getWorld().spawn(player.getEyeLocation(), TextDisplay.class);
	Bukkit.getScheduler().scheduleSyncDelayedTask(this, () -> {
		player.sendMessage(Component.text(display.getInterpolationDuration()));
		display.setInterpolationDuration(10);
		player.sendMessage(Component.text(display.getInterpolationDuration()));
	}, 20);
}
```
I can confirm that the issue was replicated using this test too. I made sure to check the spawned entities NBT too.

Finally, to ensure good results, I once again tested this server on an entirely fresh installation. I encountered the same results as the previous two tests.

Here is additional information as well:
```
/plugins output
[11:10:48 INFO]: Server Plugins (1):
[11:10:48 INFO]: Paper Plugins:
[11:10:48 INFO]:  - PickleTestPlugin

/version output
[11:10:56 INFO]: Checking version, please wait...
[11:10:57 INFO]: This server is running Paper version git-Paper-451 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 85a67da)
You are running the latest version

NBT Check Command
/data get entity @e[limit=1,type=minecraft:text_display]

java --version
openjdk 17.0.3 2022-04-19
OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode, sharing)

Tested on Vanilla Minecraft 1.19.4 profile
```

I'm not really sure what is going on here then, but if there is any additional information I can provide, please let me know.

**Comment by J-ak-e** [comment]
> Hello, I was working with Jake to determine the cause of the issue.
> 
> My initial test was conducted on a small test server with a few various plugins. I was able to reproduce the issue. Admittedly, not a very "clean" environment, so I performed some additional tests.
> 
> My next test was to remove all plugins and just use a simple testing plugin (paper-plugin specifically), similar to the provided above. Here is the code I used:
> ```java
> @EventHandler
> public void on(final PlayerDropItemEvent event) {
> 	final Player player = event.getPlayer();
> 
> 	// before actual spawn
> 	player.getWorld().spawn(player.getEyeLocation(), TextDisplay.class, display -> {
> 		player.sendMessage(Component.text(display.getInterpolationDuration()));
> 		display.setInterpolationDuration(10);
> 		player.sendMessage(Component.text(display.getInterpolationDuration()));
> 	});
> 
> 	// spawn but don't modify until after
> 	final TextDisplay display = player.getWorld().spawn(player.getEyeLocation(), TextDisplay.class);
> 	Bukkit.getScheduler().scheduleSyncDelayedTask(this, () -> {
> 		player.sendMessage(Component.text(display.getInterpolationDuration()));
> 		display.setInterpolationDuration(10);
> 		player.sendMessage(Component.text(display.getInterpolationDuration()));
> 	}, 20);
> }
> ```
> I can confirm that the issue was replicated using this test too. I made sure to check the spawned entities NBT too.
> 
> Finally, to ensure good results, I once again tested this server on an entirely fresh installation. I encountered the same results as the previous two tests.
> 
> Here is additional information as well:
> ```
> /plugins output
> [11:10:48 INFO]: Server Plugins (1):
> [11:10:48 INFO]: Paper Plugins:
> [11:10:48 INFO]:  - PickleTestPlugin
> 
> /version output
> [11:10:56 INFO]: Checking version, please wait...
> [11:10:57 INFO]: This server is running Paper version git-Paper-451 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 85a67da)
> You are running the latest version
> 
> NBT Check Command
> /data get entity @e[limit=1,type=minecraft:text_display]
> 
> java --version
> openjdk 17.0.3 2022-04-19
> OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
> OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode, sharing)
> 
> Tested on Vanilla Minecraft 1.19.4 profile
> ```
> 
> I'm not really sure what is going on here then, but if there is any additional information I can provide, please let me know.

♥️

**Comment by lynxplay** [comment]
A quick documentation purpose update on this issue (as most of the relevant discussion was on the discord).
This issue boils down to a reobf/general mapping issue introduced somewhat by upstream by making previously private methods public which causes an unwanted override.

Specifically these two mappings in question:
```
net.minecraft.world.entity.Display -> bfg:
	695:696:void setLineWidth(int) -> b
net.minecraft.world.entity.Display$TextDisplay -> bfg$k:
	345:346:void setInterpolationDuration(int) -> b
```
map private methods which spigot at'ed to public, causing `setLineWidth(int) ` to override `setInterpolationDuration(int)`.

TLDR; Hopefully we should have a fix for this sooner than later.
We may also be able to improve our tooling to prevent such issues / find out about such issues.
